src/server/pps/server/api_server.go

```diff
@@ -712,6 +712,37 @@ func (a *apiServer) ListJob(ctx context.Context, request *ppsclient.ListJobReque
 	}, nil
 }
 
+func (a *apiServer) DeleteJob(ctx context.Context, request *ppsclient.DeleteJobRequest) (response *google_protobuf.Empty, err error) {
+	func() { a.Log(request, nil, nil, 0) }()
+	metricsFn := metrics.ReportUserAction(ctx, a.reporter, "DeleteJob")
+	defer func(start time.Time) { metricsFn(start, err) }(time.Now())
+
+	pfsAPIClient, err := a.getPfsClient()
+	if err != nil {
+		return nil, err
+	}
+	persistClient, err := a.getPersistClient()
+	if err != nil {
+		return nil, err
+	}
+	jobInfo, err := persistClient.InspectJob(ctx, &ppsclient.InspectJobRequest{
+		Job: request.Job,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if err := a.deleteJob(ctx, jobInfo); err != nil {
+		return nil, err
+	}
+	if _, err := pfsAPIClient.DeleteRepo(ctx, &pfsclient.DeleteRepoRequest{Repo: jobInfo.OutputCommit.Repo}); err != nil {
+		return nil, err
+	}
```

Comment:
I think we only want to delete the repo if the job was a singleton and not run as part of a pipeline (ie if `jobInfo.PipelineName == ""`).